assets/js/collaborative-editor/utils/workflowGraph.ts

```diff
@@ -518,3 +518,49 @@ export function findFirstJobFromTrigger(
   const edge = edges.find(e => e.source_trigger_id === triggerId);
   return edge?.target_job_id || undefined;
 }
+
+// gets job ordinals relative to start(trigger-node)
+export function getJobOrdinals(adj: Record<string, string[]>, start: string) {
+  const ordinals: Record<string, number> = { [start]: 1 };
+  const visited = new Set();
+  const queue = [start];
+
+  visited.add(start);
+
+  while (queue.length > 0) {
+    const node = queue.shift(); // pop from front
+    if (!node) continue;
+
+    const neighbors = adj[node] || [];
+    for (const next of neighbors) {
+      if (!visited.has(next)) {
+        const curr = ordinals[next] || Infinity;
+        ordinals[next] = Math.min(ordinals[node], curr) + 1;
```

Comment:
I think this `Math.min` here is unnecessary complexity. Normally BFS explores nodes level-by-level. The first time we encounter a node, we've taken the shortest path to it. Subsequent paths would be longer. We could simpliy this block by this, right ?
```javascript
if (!visited.has(next)) {
    ordinals[next] = ordinals[node] + 1;
    visited.add(next);
    queue.push(next);
}
```